src/MongoDB.Driver/Encryption/ExplicitEncryptionLibMongoCryptController.cs

```diff
@@ -219,26 +218,25 @@ public async Task<BsonValue> DecryptFieldAsync(BsonBinaryData wrappedBinaryValue
         }
 
         // private methods
-        private IKmsKeyId GetKmsKeyId(string kmsProvider, IReadOnlyList<string> alternateKeyNames, BsonDocument masterKey)
+        private KmsKeyId GetKmsKeyId(string kmsProvider, IReadOnlyList<string> alternateKeyNames, BsonDocument masterKey)
         {
             IEnumerable<byte[]> wrappedAlternateKeyNamesBytes = null;
             if (alternateKeyNames != null)
             {
                 wrappedAlternateKeyNamesBytes = alternateKeyNames.Select(GetWrappedAlternateKeyNameBytes);
             }
 
-            switch (kmsProvider)
+            if (!Enum.TryParse<KmsType>(kmsProvider, true, out var kmsType))
```

Comment:
If we decide to defer all validation on the mongocrypt level, then this validation can be removed